codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ServerDowngrader.java

```diff
@@ -78,9 +80,12 @@ protected void decode(ChannelHandlerContext ctx, Http2StreamFrame frame, List<Ob
                     out.add(full);
                 }
             } else {
-                out.add(HttpConversionUtil.toHttpRequest(id, headersFrame.headers(), validateHeaders));
+                HttpRequest req = HttpConversionUtil.toHttpRequest(id, headersFrame.headers(), validateHeaders);
+                if (!HttpUtil.isContentLengthSet(req)) {
+                    HttpUtil.setTransferEncodingChunked(req, true);
```

Comment:
Can you clarify what you mean by `that's not right`?

My suggestion was because you check `if (!HttpUtil.isContentLengthSet(req))` you don't need to call `setTransferEncodingChunked(HttpMessage m, boolean chunked)` which will try to remove the `CONTENT_LENGTH` header (bcz you know its not there), and instead just directly set the `TRANSFER_ENCODING` to `CHUNKED` here.